### PR TITLE
Support proxy for tasks related functionality

### PR DESCRIPTION
**Pull request by RomanNikitenko**
Signed-off-by: Roman Nikitenko <rnikiten@redhat.com>

### What does this PR do?
Support proxy configuration for tasks related functionality

### CODE REVIEW

**Comment by RomanNikitenko** [comment]
TODO:
comments from https://github.com/eclipse/che-theia/pull/781/ should be applied before merge the PR

**Comment by RomanNikitenko** [comment]
Tested on QE airgap cluster

test_proxy_tasks_master

**Comment by RomanNikitenko** [comment]
Tested on https://che.openshift.io/

test_proxy_tasks_master_openshift

File: plugins/task-plugin/src/machine/websocket.ts [hunk 1]
```diff
@@ -98,6 +92,46 @@ export class ReconnectingWebSocket {
...
+        const baseUrl = process.env.CHE_API;
+        const proxyUrl = process.env.http_proxy;
+
+        if (proxyUrl && proxyUrl !== '' && baseUrl) {
+            const parsedBaseUrl = url.parse(baseUrl);
+            if (parsedBaseUrl.hostname && this.shouldProxy(parsedBaseUrl.hostname)) {
+                const parsedProxyUrl = url.parse(proxyUrl);
+                const mainProxyOptions = this.getMainProxyOptions(parsedProxyUrl);
+
+                const isHttpsProxy = parsedProxyUrl.protocol && parsedProxyUrl.protocol.startsWith('https:');
```
Comment (benoitf) [inline plugins/task-plugin/src/machine/websocket.ts:108]: just a FYI, if you searched the shortest way

```typescript
 parsedProxyUrl.protocol?.indexOf('https:') === 0;
```

**Pull request merged**